File: 1669-merge-in-between-linked-lists/1669-merge-in-between-linked-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        dummy = ListNode(-1, list1)
        first, second = dummy, dummy
        
        for _ in range(b - a + 1):
            second = second.next
            
        for _ in range(a):
            first = first.next
            second = second.next
            
        first.next = list2
        while first.next:
            first = first.next
            
        first.next = second.next
        
        return dummy.next

File: 1669-merge-in-between-linked-lists/1669-merge-in-between-linked-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {number} a
 * @param {number} b
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeInBetween = function(list1, a, b, list2) {
  const dummy = new ListNode(-1, list1);
  let [first, second] = [dummy, dummy];
  
  for (let i = 0; i <= b - a; i++) second = second.next;
  
  for (let i = 0; i < a; i++) {
    first = first.next;
    second = second.next;
  }
  
  first.next = list2;
  while (first.next) first = first.next;
  
  first.next = second.next;
  second.next = null;
  
  return dummy.next;
};